Sergio ortiz development

Ortiz-Sergio opened the pull request.

Ortiz-Sergio committed:
Allows users to input comments into the website

Ortiz-Sergio committed:
Allows user to enter their comment into the text box

Ortiz-Sergio committed:
Fixes html file format by removing/adding indentations

Ortiz-Sergio committed:
Removes an unused function

abhijitnukalapati commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -32,17 +32,28 @@
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      String comment = getParameter(request, "comment-input", "");
+      commentList.add(comment);
```
do we want to filter out empty comments?

Ortiz-Sergio replied:
Good idea! Maybe also filter out the placeholder comment on the text box 'New Comment'

abhijitnukalapati commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -32,17 +32,28 @@
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
```
_minor_ - it seems a bit odd that we have a return block in the comment section with no description. How about just having a description and do away with the `@return`. What do you think?

Ortiz-Sergio replied:
Makes sense!

Ortiz-Sergio committed:
Improves security by only adding 'valid' comments to the website

Ortiz-Sergio commented:
I decided to only filter out unwanted comments, without giving the user a chance to enter their comment again since the website reloads after entering a comment anyways, so users can try again. Tell me if anything about this approach isn't valid, or if there is a better way to do it.

Ortiz-Sergio committed:
Adds clarity to documentation by removing a strange @return tag

Ortiz-Sergio committed:
Allows portfolio to use Datastore

Ortiz-Sergio committed:
Adds the comments to a datastore

abhijitnukalapati commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -52,4 +75,20 @@ private String convertToJsonUsingGson(List<String> comments) {
+   * Checks if the user's comment is not empty or the placeholder text
+   */
+  public boolean isValid(String comment) {
+      if (comment.isEmpty()) {
+          return false;
+      }
+      if (comment.isBlank()) {
+          return false;
+      }
+      if (comment.equalsIgnoreCase("New Comment")) {
+          return false;
+      }
+      return true;  
```
minor - you could also just do something succinct:
`return !comment.isEmpty() && !comment.isBlank() && !comment.equalsIgnoreCase("New Comment")`

Ortiz-Sergio replied:
True!

abhijitnukalapati commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -28,21 +31,41 @@
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      String comment = getParameter(request, "comment-input", "");
+      if(isValid(comment)) {
```
come to think of it, can we do some input validation in the UI? That way we have a more responsive web app

Ortiz-Sergio replied:
Sounds good! I'll display a little message prompting the user to try again

abhijitnukalapati commented on portfolio/src/main/webapp/index.html:
```diff
@@ -195,9 +195,20 @@ <h2 style="color:FloralWhite; background-color:green;">
+
       <div style="border: 1px solid">
-      <button onclick="getServerComments()">Comments</button>
+        <button onclick="getServerComments()">Comments</button>
```
_nitpick_ - remove indentation (this is admittedly minor, but as these keep adding up, the code can start to look messy)

Ortiz-Sergio replied:
Yes! I also dislike these but I am having trouble spotting them on the editor for some reason

Ortiz-Sergio committed:
Adds clarity to code by condensing function

Ortiz-Sergio committed:
Fixes html formatting by removing indentation

Ortiz-Sergio committed:
Adds Security by validating the user's input

Ortiz-Sergio commented:
I added some input validation where a little text box pops up if the user enters an invalid comment. I added it on the html though so I am not sure if it is overkill to do validation both on the html and on the java servlet. If so, I could remove the one on teh java servlet since by the time a comment gets in it will already be a valid one.

Ortiz-Sergio committed:
Adds clarity by giving a variable a more descriptive name

Ortiz-Sergio committed:
Stores comments in a dedicated Comment class

Ortiz-Sergio committed:
References the 'comment' attribute of the class

Ortiz-Sergio committed:
Allows for the retrieval of comments from the datastore

abhijitnukalapati commented:
> I added some input validation where a little text box pops up if the user enters an invalid comment. I added it on the html though so I am not sure if it is overkill to do validation both on the html and on the java servlet. If so, I could remove the one on teh java servlet since by the time a comment gets in it will already be a valid one.


That's true. Good point. We can remove the validation on the servlet

abhijitnukalapati commented on portfolio/src/main/webapp/index.html:
```diff
@@ -195,7 +195,29 @@ <h2 style="color:FloralWhite; background-color:green;">
+          var isBlank = comment.toUpperCase() === "NEW COMMENT";
+          if (isBlank || !comment.replace(/\s/g, '').length) {
+              alert("Not a valid comment, try again");
+              return false;
```
looks good!
